Common/Core/PID/TPCPIDResponse.h

```diff
@@ -0,0 +1,216 @@
+    reso >= 0.f ? resolution = reso : resolution = 0.f;
+  } else {
+
+    std::vector<double> values;
+    const double ncl = track.tpcNClsFound();
+    const double p = track.tpcInnerParam();
+    const double mass = o2::track::pid_constants::sMasses[id];
+    const double bg = p / mass;
+    const double dEdx = mMIP * o2::tpc::BetheBlochAleph((float)bg, mBetheBlochParams[0], mBetheBlochParams[1], mBetheBlochParams[2], mBetheBlochParams[3], mBetheBlochParams[4]) * std::pow((float)o2::track::pid_constants::sCharges[id], mChargeFactor);
+    const double relReso = o2::pid::tpc::Response::GetRelativeResolutiondEdx(p, mass, o2::track::pid_constants::sCharges[id], mResolutionParams[mReadOutChamber][3]);
+
+    values.push_back((1. / dEdx));
+    values.push_back((track.tgl()));
+    values.push_back((std::sqrt(mMaxClusters / ncl)));
+    values.push_back(relReso);
+    values.push_back((track.signed1Pt()));
+    values.push_back((collision.multTPC()) / mMultNormalization);
+
+    fSigmaParametrization->SetParameters(mResolutionParams[mReadOutChamber].data());
+
+    resolution = fSigmaParametrization->EvalPar(values.data()) * (dEdx);
+  }
+  return resolution;
+}
+
+/// Gets the number of sigma between the actual signal and the expected signal
+template <typename CollisionType, typename TrackType>
+inline float Response::GetNumberOfSigma(const CollisionType& collision, const TrackType& trk, const o2::track::PID::ID id) const
+{
+  return ((trk.tpcSignal() - GetExpectedSignal(trk, id)) / GetExpectedSigma(collision, trk, id));
+}
+
+/// Gets the deviation between the actual signal and the expected signal
+template <typename TrackType>
+inline float Response::GetSignalDelta(const TrackType& trk, const o2::track::PID::ID id) const
+{
+  return (trk.tpcSignal() - GetExpectedSignal(trk, id));
+}
+
+//// Gets relative dEdx resolution contribution due relative pt resolution
+inline float Response::GetRelativeResolutiondEdx(const float p, const float mass, const float charge, const float resol) const
+{
+  const float bg = p / mass;
+  const float dEdx = o2::tpc::BetheBlochAleph(bg, mBetheBlochParams[0], mBetheBlochParams[1], mBetheBlochParams[2], mBetheBlochParams[3], mBetheBlochParams[4]) * std::pow(charge, mChargeFactor);
+  const float deltaP = resol * std::sqrt(dEdx);
+  const float bgDelta = p * (1 + deltaP) / mass;
+  const float dEdx2 = o2::tpc::BetheBlochAleph(bgDelta, mBetheBlochParams[0], mBetheBlochParams[1], mBetheBlochParams[2], mBetheBlochParams[3], mBetheBlochParams[4]) * std::pow(charge, mChargeFactor);
+  const float deltaRel = TMath::Abs(dEdx2 - dEdx) / dEdx;
+  return deltaRel;
+}
+
+inline void Response::PrintAll() const
+{
+  LOGP(info, "==== TPC PID response parameters: ====");
+  for (int i = 0; i < int(mBetheBlochParams.size()); i++)
+    LOGP(info, "BB param [{}] = {}", i, mBetheBlochParams[i]);
```

Comment:
Please always use readability braces:
```suggestion
  for (int i = 0; i < int(mBetheBlochParams.size()); i++) {
    LOGP(info, "BB param [{}] = {}", i, mBetheBlochParams[i]);
  }
```